packages/components/managed-data-table/src/data-table-settings/data-table-settings.js

```diff
@@ -0,0 +1,189 @@
+import PropTypes from 'prop-types';
+import React, { useState } from 'react';
+import requiredIf from 'react-required-if';
+import { useIntl } from 'react-intl';
+import Constraints from '@commercetools-uikit/constraints';
+import SelectInput from '@commercetools-uikit/select-input';
+import Spacings from '@commercetools-uikit/spacings';
+import DensityManager from './density-manager';
+import ColumnManager from './column-manager';
+import TABLE_SETTINGS_ACTIONS, {
+  COLUMN_MANAGER,
+  DISPLAY_SETTINGS,
+} from '../constants';
+import messages from './messages';
+
+export const getDropdownOptions = ({
+  isColumnManagerEnabled,
+  isDisplaySettingsEnabled,
+  formatMessage,
+}) => [
+  ...(isColumnManagerEnabled
+    ? [
+        {
+          value: COLUMN_MANAGER,
+          label: formatMessage(messages.columnManagerOption),
+        },
+      ]
+    : []),
+  ...(isDisplaySettingsEnabled
+    ? [
+        {
+          value: DISPLAY_SETTINGS,
+          label: formatMessage(messages.displaySettingsOption),
+        },
+      ]
+    : []),
+];
+
+export const getMappedColumns = (columns = []) =>
+  columns.reduce(
+    (mappedColumns, column) => ({
+      ...mappedColumns,
+      [column.key]: column,
+    }),
+    {}
+  );
+
+export const getSelectedColumns = (visibleColumnsKeys = [], mappedColumns) =>
+  visibleColumnsKeys.map((columnKey) => mappedColumns[columnKey]);
+
+const DataTableSettings = (props) => {
+  const intl = useIntl();
+  const [openPanel, setOpenPanel] = useState(null);
+
+  const dropdownOptions = getDropdownOptions({
+    isColumnManagerEnabled: !props.columnManager.disableColumnManager,
+    isDisplaySettingsEnabled: !props.displaySettings.disableDisplaySettings,
+    formatMessage: intl.formatMessage,
+  });
+
+  const handleDropdownChange = (event) => {
+    if (event.target.value === COLUMN_MANAGER) {
+      setOpenPanel(COLUMN_MANAGER);
+      return;
+    }
+    setOpenPanel(DISPLAY_SETTINGS);
+  };
+
+  const mappedColumns = getMappedColumns(props.columnManager.hideableColumns);
+
+  const selectedColumns = getSelectedColumns(
+    props.columnManager.visibleColumnKeys,
+    mappedColumns
+  );
+
+  return (
+    <Spacings.Stack>
+      <Spacings.Inline justifyContent="space-between" alignItems="center">
+        {props.topBar || <div />}
```

Comment:
Why the empty `div` as fallback?

Reply:
Because if topBar is not set we still need the dropdown to be at the right

Reply:
Should we then always wrap the `topBar` with a `<div>`? It would be good in case the `topBar` is a string and not a React component.

```suggestion
        <div>{props.topBar}</div>
```